kernel/packages/shared/atlas/reducer.ts

```diff
@@ -1,25 +1,36 @@
 import { AnyAction } from 'redux'
-import { REPORTED_SCENES_FOR_MINIMAP } from './actions'
+import { ILand } from 'shared/types'
+import { REPORTED_SCENES_FOR_MINIMAP, FetchDataFromSceneJsonSuccess, QuerySceneData, FetchDataFromSceneJsonFailure, ReportedScenes } from './actions'
+import { getSceneNameFromAtlasState, getSceneNameWithMarketAndAtlas, postProcessSceneName } from './selectors'
+// @ts-ignore
+import defaultLogger from '../logger'
 import {
   AtlasState,
-  District,
   DISTRICT_DATA,
-  FAILURE_NAME_FROM_SCENE_JSON,
-  FETCH_NAME_FROM_SCENE_JSON,
+  FAILURE_DATA_FROM_SCENE_JSON,
+  MapSceneData,
   MarketData,
   MARKET_DATA,
-  SUCCESS_NAME_FROM_SCENE_JSON
+  SUCCESS_DATA_FROM_SCENE_JSON,
+  QUERY_DATA_FROM_SCENE_JSON
 } from './types'
-import { zip } from './zip'
-import { PayloadAction } from 'typesafe-actions'
 
 const ATLAS_INITIAL_STATE: AtlasState = {
-  marketName: {},
-  sceneNames: {},
-  requestStatus: {},
-  districtName: {},
-  alreadyReported: {},
-  lastReportPosition: { x: -9999, y: -9999 }
+  hasMarketData: false,
+  hasDistrictData: false,
+  tileToScene: {}, // '0,0' -> sceneId. Useful for mapping tile market data to actual scenes.
+  idToScene: {}, // sceneId -> MapScene
+  lastReportPosition: undefined
+}
+
+const MAP_SCENE_DATA_INITIAL_STATE: MapSceneData = {
+  sceneId: '',
+  name: '',
+  type: 0,
+  estateId: 0,
+  sceneJsonData: undefined,
+  alreadyReported: false,
+  requestStatus: 'loading'
 }
 
 export function atlasReducer(state?: AtlasState, action?: AnyAction) {
```

Comment:
if we switch `AnyAction` to `QuerySceneData|FetchDataFromSceneJsonSuccess|FetchDataFromSceneJsonFailure| ReportedScenes` I believe the type casts on each `case` are not going to be necessary